caas/containers.go

```diff
@@ -50,13 +50,39 @@ type ContainerSpec struct {
 
 	// ProviderContainer defines config which is specific to a substrate, eg k8s
 	ProviderContainer `yaml:"-"`
+
+	Constraints []Constraint `yaml:"constraint,omitempty"`
```

Comment:
This doesn't belong here - this ContainerSpec struct is for processing the yaml snippet passed in by the charm via the set pod spec hook command